File: app/src/main/java/com/soobinpark/appcraft/readingbible/domain/model/BibleDiagnostics.kt
package com.soobinpark.appcraft.readingbible.domain.model

enum class BibleFileIssueSeverity {
    Info,
    Warning,
    Error,
}

data class BibleFileIssue(
    val title: String,
    val detail: String,
    val severity: BibleFileIssueSeverity,
)

data class BibleFileDiagnostic(
    val rootLabel: String,
    val scannedFileCount: Int,
    val lfaCount: Int,
    val lfbCount: Int,
    val bdfFileCount: Int,
    val completeBdfVersionCount: Int,
    val unknownFileCount: Int,
    val versions: List<BibleVersion>,
    val issues: List<BibleFileIssue>,
)

File: ios/Sources/Domain/Models/BibleDiagnostics.swift
import Foundation

enum BibleFileIssueSeverity: Sendable {
    case info
    case warning
    case error
}

struct BibleFileIssue: Identifiable, Sendable {
    let id = UUID()
    let title: String
    let detail: String
    let severity: BibleFileIssueSeverity
}

struct BibleFileDiagnostic: Sendable {
    let rootLabel: String
    let scannedFileCount: Int
    let lfaCount: Int
    let bdfFileCount: Int
    let completeBdfVersionCount: Int
    let mp3Count: Int
    let unknownFileCount: Int
    let versions: [BibleVersion]
    let issues: [BibleFileIssue]
}
